app/scripts/views/settings.js

```diff
@@ -60,6 +62,22 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       this._subPanels = options.subPanels || this._initializeSubPanels(options);
       this._formPrefill = options.formPrefill;
       this.router.on(this.router.NAVIGATE_FROM_SUBVIEW, this._onNavigateFromSubview.bind(this));
+
+      var self = this;
+      // Statically bound so that it can be passed to interTabOn/Off.
+      this._clearSessionAndNavigateToSignin = function () {
+        self.logScreenEvent(Constants.SIGNOUT_SUCCESS);
+        self.user.clearSignedInAccount();
+        // The user has manually signed out, a pretty strong indicator
+        // the user does not want any of their information pre-filled
+        // on the signin page. Clear any remaining formPrefill info
+        // to ensure their data isn't sticking around in memory.
+        self._formPrefill.clear();
+        Session.clear();
+        self.navigate('signin', {
+          success: t('Signed out successfully')
+        });
+      };
```

Comment:
nit: initialize looks overloaded, anyway to move this diff out?  cc @shane-tomlinson